src/clients/c++/perf_client/perf_client.cc

```diff
@@ -770,14 +795,34 @@ main(int argc, char** argv)
         } else if (arg.compare("constant") == 0) {
           request_distribution = Distribution::CONSTANT;
         } else {
-          Usage(argv, "unsupported input data provided " + std::string(optarg));
+          Usage(
+              argv, "unsupported request distribution provided " +
+                        std::string(optarg));
         }
         break;
       }
       case 20:
         using_custom_intervals = true;
         request_intervals_file = optarg;
         break;
+      case 21: {
+        std::string arg = optarg;
+        if (arg.compare("system") == 0) {
```

Comment:
In windows, shared memory is not supported. This means that build for the perf client should change accordingly and an error should be thrown if the client tries to run the perf client with shared memory.

Reply:
We currently don't build our examples and perf_client on windows. Probably, will file a ticket to add the support.

Reply:
Yes you can do that